src/DistributedTracingManager.ts

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as vscode from "vscode";
+import { BaseExplorer } from "./baseExplorer";
+import { Constants, DistributedSettingUpdateType } from "./constants";
+import { TelemetryClient } from "./telemetryClient";
+import { Utility } from "./utility";
+import iothub = require("azure-iothub");
+import uuid = require("uuid");
+import { DeviceItem } from "./Model/DeviceItem";
+import { SamplingModeItem } from "./Model/SamplingModeItem";
+import { DistributedTracingLabelNode } from "./Nodes/DistributedTracingLabelNode";
+import { DistributedTracingSettingNode } from "./Nodes/DistributedTracingSettingNode";
+
+export class DistributedTracingManager extends BaseExplorer {
+    constructor(outputChannel: vscode.OutputChannel) {
+        super(outputChannel);
+    }
+
+    public async updateDistributedTracingSetting(node, updateType?: DistributedSettingUpdateType) {
+        let iotHubConnectionString = await Utility.getConnectionString(Constants.IotHubConnectionStringKey, Constants.IotHubConnectionStringTitle);
+        if (!iotHubConnectionString) {
+            return;
+        }
+
+        if (!updateType) {
+            updateType = DistributedSettingUpdateType.All;
+        }
+
+        let deviceIds: string[];
+        if (!node) {
+            const deviceIdList = await Utility.getNoneEdgeDeviceIdList(iotHubConnectionString);
+            const deviceItemList = deviceIdList.map((deviceId) => new DeviceItem(deviceId, null, null, null, null));
+            let selectedDevices: DeviceItem[] = await vscode.window.showQuickPick(
+                deviceItemList,
+                { placeHolder: "Select device...", ignoreFocusOut: true, canPickMany: true },
+            );
+            deviceIds = selectedDevices.map((deviceItem) => deviceItem.deviceId);
+        } else {
+            deviceIds = [node.deviceNode.deviceId];
+        }
+
+        this._outputChannel.show();
+        await this.updateDistributedTracingSettingForDevices(deviceIds, iotHubConnectionString, updateType);
+
+        if (node instanceof DistributedTracingLabelNode) {
```

Comment:
If user cancel, should not refresh

Reply:
fixed

Reply:
If user cancel, still refresh, please check